### ISSUE TITLE
Nested `list()` variables reported as undefined

### ISSUE THREAD

**Issue by dlh01** [Automattic/VIP-Coding-Standards#458]
## Bug Description

Variables assigned in a nested call to `list()` are reported as undefined.

## Minimal Code Snippet

```php
list(
	$foo,
	list(
		$bar,
	)
) = [ 'foo', [ 'bar' ] ];
```

## Error Code

`Variable $bar is undefined. (WordPressVIPMinimum.Variables.VariableAnalysis.UndefinedVariable)`

## Environment

| Question                | Answer
| ------------------------| -------
| PHP version             | 7.3.12
| PHP_CodeSniffer version | 3.5.1
| VIPCS version           | 2.0.0

## Tested Against `master` branch?

- [x] I have verified the issue still exists in the `master` branch of VIPCS.

**Comment by GaryJones** [comment]
@sirbrillig Do you know if this test case is addressed in your fork of VariableAnalysis?

**Comment by sirbrillig** [comment]
Ohhh. I hadn't considered that. It does not, but I've created a (failing) test case so this should be solvable: https://github.com/sirbrillig/phpcs-variable-analysis/pull/129

**Comment by GaryJones** [comment]
Thanks @sirbrillig.

@dlh01 I'll close this one for now then, as the issue will be fixed once we switch to using @sirbrillig's Variable Analysis package (#449) rather than the forked sniff we have at the moment.

**Comment by GaryJones** [comment]
Now that #449 is completed to use VariableAnalysis proper, I've confirmed that the provided example above doesn't give an error with the `develop` version of `WordPress-VIP-Go`, so this is now fixed for VIPCS 2.2.0.

**Comment by GaryJones** [comment]
This is fixed in the 3.x branch of VariableAnalysis.

VIPCS 2.1 (released) uses the old broken version of a fork of VA.
VIPCS 2.2 (in development) currently supports VA 2.x, but depending on timelines, we may be able to support / require 3.x as well/instead. 

In short, we need to wait for VA 3.x to be released before we can make a release of VIPCS that supports it which will then address the issue reported here.

I've added our 2.2.0 milestone as a reminder to we can keep an eye on how VA 3.x is progressing :-)
